### ISSUE TITLE
Redirect from middleware hook

### ISSUE THREAD

**Issue by bbuhler** [Elderjs/elderjs#185]
Hello 👋️

I try to perform an HTTP redirect from a middleware hook based on route request object.

```js
  {
    hook: 'middleware',
    name: 'redirect',
    description: 'Handle route redirects',
    priority: 60,
    run({ res, req, serverLookupObject, next })
    {
      const lookedupRequest = serverLookupObject[req.url];

      if (lookedupRequest?.redirectTo)
      {
        res.redirect(lookedupRequest.redirectTo);
        res.headerSent = true;
        // next();
      }

      return { res };
    }
  },
```

It somehow works, but I get 'Cannot set headers after they are sent to the client' errors in the logs.

How can I prevent that the subsequent hooks and rendering is being executed?
Is there a smarter way to archive that?

**Comment by nickreese** [comment]
Just shipped a fix for this in `v1.4.5`.

Here is the hook that was working as expected (polka specific not express):

```js
  {
    hook: 'middleware',
    name: 'redirect',
    description: 'Handle route redirects',
    priority: 60,
    run({ res, req, redirect, next }) {
      if (req.path.includes('redirect')) {
        const str = `Redirecting to ${`http://example.com`}`;
        console.log(str);
        res.writeHead(302, {
          Location: `http://example.com`,
          'Content-Type': 'text/plain',
          'Content-Length': str.length,
        });
        res.end(str);
      }
    },
  },
```

**Comment by nickreese** [comment]
@bbuhler Looks like this was either a typo or I was referencing old express docs. Either way, it now checks for plural and singular.
